ui/src/main/webapp/src/app/components/reports/source/source-report.component.ts

```diff
@@ -0,0 +1,283 @@
+import {InlineHintModel} from "../../../generated/tsModels/InlineHintModel";
+import {HintService} from "../../../services/graph/hint.service";
+import {LinkModel} from "../../../generated/tsModels/LinkModel";
+import {SourceFileModel} from "../../../generated/tsModels/SourceFileModel";
+import {GraphJSONToModelService} from "../../../services/graph/graph-json-to-model.service";
+import {Http} from "@angular/http";
+import {LinkableModel} from "../../../generated/tsModels/LinkableModel";
+import {Observable} from "rxjs";
+
+@Component({
+    templateUrl: '/source-report.component.html',
+    styleUrls: [ '/source-report.component.css' ]
+})
+export class SourceReportComponent implements OnInit, AfterViewChecked {
+    private execID: number;
+    private fileID: number;
+    private fileSource: string = "Loading...";
+    private fileLines: string[];
+
+    private fileModel: FileModel;
+    private sourceFileModel: SourceFileModel;
+    private transformedLinks: LinkModel[];
+
+    private classifications: ClassificationModel[];
+    private classificationLinks: Map<ClassificationModel, LinkModel[]> = new Map<ClassificationModel, LinkModel[]>();
+    private hints: InlineHintModel[];
+    private rendered: boolean = false;
+
+    constructor(private route:ActivatedRoute,
+                private fileModelService:FileModelService,
+                private classificationService:ClassificationService,
+                private hintService:HintService,
+                private http:Http
+    ) { }
+
+    ngOnInit(): void {
+        this.route.params.forEach((params: Params) => {
+            this.execID = +params['executionId'];
+            this.fileID = +params['fileId'];
+
+            this.fileModelService.getFileModel(this.execID, this.fileID)
+                .subscribe((fileModel) => {
+                    this.fileModel = fileModel;
+
+                    // Assume this is a source file model and deserialize it as one... if it is not, this will have a lot of null
+                    //   properties
+                    this.sourceFileModel = <SourceFileModel>new GraphJSONToModelService().fromJSON(this.fileModel.data, this.http, SourceFileModel);
+                    this.sourceFileModel.linksToTransformedFiles.subscribe((links) => this.transformedLinks = links);
+                });
+
+            this.classificationService.getClassificationsForFile(this.execID, this.fileID)
+                .subscribe((classifications) => this.classifications = classifications );
+
+            this.hintService.getHintsForFile(this.execID, this.fileID)
+                .subscribe((hints) => this.hints = hints);
+
+            this.fileModelService.getSource(this.execID, this.fileID)
+                .subscribe((fileSource) => {
+                    this.fileSource = fileSource;
+                    this.fileLines = fileSource.split(/[\r\n]/).map((line) => line + "\n");
+                });
+        });
```

Comment:
Please add some error handling. I think adding notification service and creating error notification when some of those subscribe calls fails should be enough.